### PR TITLE
[10.0][FIX] stock_scanner: Improve login method

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

File: stock_scanner/models/scanner_hardware.py [hunk 1]
```diff
@@ -451,6 +451,8 @@ def scanner_end(self, numterm=None):
 
     @api.one
     def check_credentials(self, login, password):
+        if password == '':
```
Comment (sbidoul) [inline stock_scanner/models/scanner_hardware.py]:
```suggestion
        if not password:
```

**Review by jgrandguillaume — APPROVED** [review]
Hi,

Thx for the contrib. LGTM if you changed what's Stéphane pointed out.

Regards,

Joël

File: stock_scanner/models/scanner_hardware.py (hunk 1 above)
Comment (rousseldenis) [reply stock_scanner/models/scanner_hardware.py]: Ok! It was to highlight the use case but I change it

Comment (rousseldenis) [reply stock_scanner/models/scanner_hardware.py]: Done!

**Review by benwillig — APPROVED** [review]

**Review by lmignon — APPROVED** [review]

**Review by dreispt — APPROVED** [review]

**Review by lmignon — APPROVED** [review]
Thank you @rousseldenis

**Comment by lmignon** [comment]
@rousseldenis Nice your last test.. travis is :red_circle:  :smirk:

**Comment by rousseldenis** [comment]
needs fixing. Want to add a test

**Comment by rousseldenis** [comment]
Done !

**Comment by pedrobaeza** [comment]
Should this be forward-ported to v11?

**Comment by rousseldenis** [comment]
@pedrobaeza No. The problem is fixed in Odoo (I made the OCA PR as I didn't know how long that could take) since https://github.com/odoo/odoo/commit/6639630de11e5e3b4046968f8981415d6ccca6cd

**Pull request merged**